### DISCUSSION TITLE
How to download a folder?

### DISCUSSION THREAD

**Discussion by heguixin** [gildas-lormeau/zip.js#348] [Q&A]
It's hard for me to learn how to use zip.js, wuwu~.
Could you give me an example to solve this problom?
eg:
you have a file dict:
```
{
     1:"http://xxxxx/asdffdafdsfs"
     2:"http://xxxxx/asdfadsdsad"
     3:"http://xxxxx/asddsadsaad"
}

```
another dict:(stand for the per file's path)
```
{
     1:"dir1/"
     2:"dir1/dir1-1"
     3:"dir1/dir1-2"
}
```

Finally, you should download  all files  to a zip .

**Comment by gildas-lormeau** [comment]
The example you provided is not clear enough to understand what you want to achieve precisely. For example, a "dict" does not exist in JS. I guess you're referring to an object even if the structure looks like an array? I guess "dir1/" is a folder and I guess "asdffdafdsfs" is a simple file (not a folder, not a zip)? I also guess "dir1/dir1-1" and "dir1/dir1-2" are folders even there isn't any ending "/"? That's a lot of ambiguity for a simple question...

I guess you want something similar to this: https://jsfiddle.net/8196k3j4

```js
import * as zip from "https://unpkg.com/@zip.js/zip.js@2.6.12/index.js";

main();

async function main() {
	const urls = {
		1: "https://unpkg.com/@zip.js/zip.js@2.6.12/tests/all/test-blob.js",
		2: "https://unpkg.com/@zip.js/zip.js@2.6.12/tests/all/test-base64.js",
		3: "https://unpkg.com/@zip.js/zip.js@2.6.12/tests/all/test-array.js"
	};
	const filenames = {
		1: "dir1/",
		2: "dir1/dir1-1/",
		3: "dir1/dir1-2/"
	};
  
	// Creates the zip file
	const zipWriter = new zip.ZipWriter(new zip.BlobWriter("application/zip"));
	await Promise.all(Object.keys(urls).map(async key => {
		const url = urls[key];
		const filename = filenames[key] + url.split("/").pop();
		await zipWriter.add(filename, new zip.HttpReader(url));
	}));

	// Downloads the Blob object containing the zip file.
	const anchorElement = document.createElement("a");
	anchorElement.href = URL.createObjectURL(await zipWriter.close());
	anchorElement.download = "test.zip";
	anchorElement.click();
}
```
